### PR TITLE
Attachment Upload

### PR DESCRIPTION
**Pull request by chentsulin**

### CODE REVIEW

**Comment by kpman** [comment]
image

File: src/cli/providers/messenger/attachment.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      )} is under heavy development. API may change between any versions.`
+    );
+
+    const config = getConfig('bottender.config.js', 'messenger');
+
+    invariant(config.accessToken, 'accessToken is not found in config file');
+
+    const client = MessengerClient.connect(config.accessToken);
+
+    const files = await readdir('assets', [ignoreFunc]);
```
Comment (chentsulin) [inline src/cli/providers/messenger/attachment.js]:
```js
const files = await readdir('assets', ['.*']);
```

File: src/cli/providers/messenger/attachment.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import help from './help';
+
+const getFileType = file => {
+  const imageType = ['jpg', 'png', 'jpeg', 'gif'];
+  const videoType = ['avi', 'mp4', 'm4v'];
+  const audioType = ['mp3', 'mid', 'm4a', 'wav'];
+
+  let type = 'file';
+
+  const buffer = readChunk.sync(file, 0, 4100);
```
Comment (chentsulin) [inline src/cli/providers/messenger/attachment.js]: we can name 0, 4100 magic numbers here to improve readability.

File: src/cli/providers/messenger/attachment.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  const typeResult = fileType(buffer);
+  if (typeResult) {
+    const { ext } = typeResult;
+    if (imageType.includes(ext)) {
+      type = 'image';
+    } else if (videoType.includes(ext)) {
+      type = 'video';
+    } else if (audioType.includes(ext)) {
+      type = 'audio';
+    }
```
Comment (chentsulin) [inline src/cli/providers/messenger/attachment.js]: Should return file type in else branch?

File: src/cli/providers/messenger/attachment.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    return process.exit(1);
+  }
+}
+
+export default async function main(ctx) {
+  const subcommand = ctx.argv._[2];
+  switch (subcommand) {
+    case 'upload':
+      await uploadAttachment();
+      break;
```
Comment (chentsulin) [inline src/cli/providers/messenger/attachment.js]: should help subcommand work here?

File: src/cli/providers/messenger/attachment.js (hunk 3 above)
Comment (kpman) [reply src/cli/providers/messenger/attachment.js]: the `type` is defined as 'file' at the beginning.
I've adjusted the order for readability.

File: src/cli/providers/messenger/attachment.js (hunk 4 above)
Comment (kpman) [reply src/cli/providers/messenger/attachment.js]: sure

**Pull request merged**